Add an OpenTelemetry gRPC access logger (just gRPC handling)

itamarkam opened the pull request:
Commit Message: Add an [OpenTelemetry](http://opentelemetry.io) access logger (only gRPC handling)
Additional Description: This PR adds a dependency on the OpenTelemetry protos used for logging. The actual part where the log entry is filled ("Common::ImplBase::emitLog") isn't implemented yet, and there's no API to activate this logger.
Risk Level: Low (this is not linked to the Envoy binary as nothing creates this logger)
Testing: Unit tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

itamarkam commented:
/assign yanavlasov

yanavlasov commented:
@itamarkam thanks this looks good. I would suggest moving the `grpc_ot_access_log_impl.*` and corresponding tests into its own directory (i.e. `source/extensions/access_loggers/open_telemetry` or some such), since this is a distinct logger from Envoy's gRPC logger.
Please also register the new logger in the `source/extensions/extensions_build_config.bzl`

yanavlasov commented:
/wait

itamarkam commented:
> @itamarkam thanks this looks good. I would suggest moving the `grpc_ot_access_log_impl.*` and corresponding tests into its own directory (i.e. `source/extensions/access_loggers/open_telemetry` or some such), since this is a distinct logger from Envoy's gRPC logger.

Done. I also moved it from the GrpcCommon namespace to its own OpenTelemetry namespace, and removed the "OpenTelemetry" prefix from all the access logger and cache classes.

> Please also register the new logger in the `source/extensions/extensions_build_config.bzl`
This is not yet a configurable extension - it's missing the proto definition on how to configure it and the logger that actually fills a log entry based on the request/response info. I was planning on adding those 2 in a separate PR to reduce the review complexity. WDYT?

itamarkam commented:
/assign htuch for OTLP dependency approval

itamarkam commented:
> CC @envoyproxy/dependency-shepherds: Your approval is needed for changes made to `(bazel/.*repos.*\.bzl)|(bazel/dependency_imports\.bzl)|(api/bazel/.*\.bzl)|(.*/requirements\.txt)|(.*\.patch)`.

/assign htuch

moderation commented on api/bazel/repository_locations.bzl:
```diff
@@ -96,4 +96,15 @@ REPOSITORY_LOCATIONS_SPEC = dict(
+        strip_prefix = "opentelemetry-proto-{version}",
+        urls = ["https://github.com/open-telemetry/opentelemetry-proto/archive/v{version}.tar.gz"],
+        release_date = "2020-10-27",
+        use_category = ["api"],
```
`0.7.0` was released a week ago - can we move to that latest version for this PR? https://github.com/open-telemetry/opentelemetry-proto/releases/tag/v0.7.0

itamarkam replied:
Done.

moderation reviewed (commented):
Question re: including latest dependency version to prevent update work in the future

moderation commented:
/lgtm deps

itamarkam commented:
/review yanavlasov

itamarkam commented:
@moderation I think that the latest merge requires a deps approval again.

moderation commented:
/lgtm deps
